platform/linux-generic/odp_classification.c

```diff
@@ -220,10 +221,11 @@ odp_cos_t odp_cls_cos_create(const char *name, odp_cls_cos_param_t *param)
 	cos_t *cos;
 	uint32_t tbl_index;
 
-	/* Packets are dropped if Queue or Pool is invalid*/
-	if (param->num_queue > CLS_COS_QUEUE_MAX)
+	/* num_queue should not be zero */
+	if (param->num_queue > CLS_COS_QUEUE_MAX || param->num_queue < 1)
 		return ODP_COS_INVALID;
 
+	/* Packets are dropped if Queue (except hashing) or Pool is invalid */
```

Comment:
The preivous comments were above the drop policy explaining that packet will be dropped if queue or pool is invalid but since the introduction of hash queues the queue could be technically nil since the queues will be created by the implementation. Maybe I will update the comments to be more clear